se/formatting.py

```diff
@@ -50,78 +50,76 @@ def semanticate(xhtml: str) -> str:
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)Mdlle\.", r"<abbr>Mdlle.</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)Mlles\.", r"<abbr>Mlles.</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)Messrs\.", r"<abbr>Messrs.</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)Messers\.", r"<abbr>Messers.</abbr>", xhtml)
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)([Vv])ol(s?)\.", r"<abbr>\1ol\2.</abbr>", xhtml)
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)([Cc])hap\. ([0-9])", r"<abbr>\1hap.</abbr> \2", xhtml) # The number allows us to avoid phrases like `Hello, old chap.`
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>|\.)(P\.(?:P\.)?S\.(?:S\.)?\B)", r"""<abbr epub:type="z3998:initialism">\1</abbr>""", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)Co\.", r"<abbr>Co.</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)Inc\.", r"<abbr>Inc.</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)Ltd\.", r"<abbr>Ltd.</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)St\.", r"<abbr>St.</abbr>", xhtml)
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)([Gg])ov\.", r"<abbr>\1ov.</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)Col\.", r"<abbr>Col.</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)MS(S?)\.", r"""<abbr>MS\1.</abbr>""", xhtml)
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)([Vv])iz\.", r"<abbr>\1iz.</abbr>", xhtml)
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)etc\.", r"<abbr>etc.</abbr>", xhtml)
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)([Cc])f\.", r"<abbr>\1f.</abbr>", xhtml)
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)ed\.", r"<abbr>ed.</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)(Jan\.|Feb\.|Mar\.|Apr\.|Jun\.|Jul\.|Aug\.|Sep\.|Sept\.|Oct\.|Nov\.|Dec\.)", r"<abbr>\1</abbr>", xhtml)
+	xhtml = regex.sub(r"(?<!\<abbr[^\>]*?\>)No\.(\s+[0-9]+)", r"<abbr>No.</abbr>\1", xhtml)
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)([Vv])s\.", r"<abbr>\1s.</abbr>", xhtml)
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)([Ff])f\.", r"<abbr>\1f.</abbr>", xhtml) # ff. typically used in footnotes, means "and following"
+	xhtml = regex.sub(r"\b(?<!\<abbr[^\>]*?\>)([Ll])ib\.", r"<abbr>\1ib.</abbr>", xhtml) # Lib. = Liber = Book
+	xhtml = regex.sub(r"""(?<!\<abbr[^\>]*?\>)([Ii])\.e\.""", r"""<abbr epub:type="z3998:initialism">\1.e.</abbr>""", xhtml)
+	xhtml = regex.sub(r"""(?<!\<abbr[^\>]*?\>)([Ee])\.g\.""", r"""<abbr epub:type="z3998:initialism">\1.g.</abbr>""", xhtml)
+	xhtml = regex.sub(r"""(?<!\<abbr[^\>]*?\>)\bN\.?B\.\B""", r"""<abbr epub:type="z3998:initialism">N.B.</abbr>""", xhtml)
+	xhtml = regex.sub(r"""(?<!\<abbr[^\>]*?\>)Ph\.?\s*D\.?""", r"""<abbr epub:type="z3998:name-title">Ph. D.</abbr>""", xhtml)
+	xhtml = regex.sub(r"""(?<!\<abbr[^\>]*?\>)(?:IOU(?:\.|\b)|I\.O\.U\.)""", r"""<abbr epub:type="z3998:initialism">I.O.U.</abbr>""", xhtml)
+	xhtml = regex.sub(r"""(?<!\<abbr[^\>]*?\>)T\.?V\.?\b""", r"""<abbr epub:type="z3998:initialism">TV</abbr>""", xhtml)
```

Comment:
This is the same pattern as N.B. but the regex differs. Should we make them the same?

Reply:
The TV? Yes, almost certainly; I'll see if I can add a couple of cases and test that tomorrow, and look around for any others.
And I forgot something in my improvements list: The IOU regex before was creating an extra period in some cases, e.g. `<p>This is an IOU.</p>` was turned into `<p>This is an <abbr epub:type="z3998:initialism">I.O.U.</abbr>.</p> I changed it so that it left only a single period in all cases (and only allow IOU or I.O.U.). The eoc check will then pick it up (it wasn't before because of the extra period).

Reply:
Grr, I forgot to tag the "turned into." That should be "…turned into `<p>This is an <abbr epub:type="z3998:initialism">I.O.U.</abbr>.</p>`".